source/agora/network/NetworkManager.d

```diff
@@ -241,28 +241,46 @@ public class NetworkManager
 
         logInfo("Establishing connection with %s...", address);
         auto node = new NetworkClient(this.taskman, address,
-            this.getClient(address), this.node_config.retry_delay.msecs);
+            this.getClient(address), this.node_config.retry_delay.msecs,
+            this.node_config.max_retries);
 
-        node.handshake();
-        this.onNodeConnected(node);
+        while (1)
+        {
+            try
+            {
+                node.handshake();
+                this.onNodeConnected(node);
+                break;
+            }
+            catch (Exception ex)
+            {
+               // keep trying until node is connected
+            }
+        }
 
         // keep asynchronously polling for complete network info,
-        // until complete peer info is returned or the parent
-        // node has established all necessary connections
+        // until complete peer info is returned, or we've
+        // established all necessary connections,
+        // or the node was banned
         while (!this.minPeersConnected())
         {
-            NetworkInfo net_info;
-            if (node.getNetworkInfo(net_info))
+            try
             {
-                // received some network info (may still be incomplete)
+                auto net_info = node.getNetworkInfo();
                 this.addAddresses(net_info.addresses);
                 if (net_info.state == NetworkState.Complete)
-                    break;
+                    break;  // done
+            }
+            catch (Exception ex)
+            {
+                // attempt later
             }
 
+            // if it's incomplete give the client some time to connect
+            // with other peers and try again later
             logInfo("[%s] (%s): Peer info is incomplete. Retrying in %s..",
-                node.address, node.key, this.node_config.retry_delay);
-            this.taskman.wait(this.node_config.retry_delay.msecs);
+                node.address, node.key, 30.seconds);
+            this.taskman.wait(30.seconds);
```

Comment:
Is that on purpose ??

Reply:
Well yes, there is a comment right above it.

Reply:
The idea is, let the node have some time too boot. I don't know if `retry_delay` is enough, because that value should be kept fairly low (a few seconds).

Reply:
Some time /= 30 seconds. But IIRC it already retry internally and that's why the delay is so high ?

Reply:
For things like `putTransaction` I've set it to follow the configured `retry_delay`. While for the periodic `getNetworkInfo` I feel like we should wait longer for the target node to establish a connection with other nodes, before asking it for its peers again.

I'm not sure, maybe I was overthinking it and we should just use `retry_delay`? `getNetworkInfo` could be pretty expensive to call, it returns an array of all known peers.

Reply:
Yeah I'd stay let's stick to retry_delay for now, there's a few optimization we'll need to do in the future anyways.